drivers/include/periph/wdt.h

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2017 Technische Universität Berlin
+ *           (C) 2019 Inria
+ *           (C) 2019 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_wdt WDT
+ * @ingroup     drivers_periph
+ * @brief       watchdog timer peripheral driver
+ *
+ * A watchdog timer (WDT) is an electronic or software timer that is
+ * used to detect and recover from unusual or suspect behaviour as well as
+ * software malfunctions. During normal operation the application will reset
+ * or wdt_kick() the timer counter preventing it from hitting 0. If due to
+ * software or hardware error the WDT counter isn't reset this will trigger
+ * corrective actions, i.e. a system reboot or callback (not supported by all
+ * platforms). 
+ *
+ * This basic interface defines two operation modes NORMAL and WINDOWED. In
+ * NORMAL operation if the WDT counter isn't reset before TIMEOUT then a
+ * reboot/callback is triggered. In WINDOWED operation if a reset isn't triggered
+ * or if it's triggered outside of the time window a reboot is triggered.
+ *
+ * Not all platforms support stopping the WDT, in some cases once enabled it
+ * can't be stopped so verify the specific platform implementation and (some
+ * only pause/stop it in some operation modes).
+ *
+ * Since WDT upper and lower bounds are a per platform constraint they can be
+ * defined in periph_cpu.h and should be known at compile time. A distinction
+ * can be made between windowed WDT and normal WDT bounds, i.e.:
+ * 
+ * - NWDT_TIME_LOWER_BOUND & NWDT_TIME_UPPER_BOUND
+ * - WWDT_TIME_LOWER_BOUND & WWDT_TIME_UPPER_BOUND
+ *
+ * NORMAL operation
+ * ================
+ *
+ *
+ *    wdt_setup_reboot(0, MAX_TIME);
+ *    wdt_start();
+ * 
+ *    while (1) {
+ *        xtimer_usleep(MAX_TIME - VALID_TIME);
+ *        wdt_kick()
+ *    }
+ *
+ * |---------------------MAX_TIME-----------------------|
+ *
+ * |---------VALID_TIME-------|
+ *
+ *                            ^
+ *                            |
+ *                       wdt_kick()
+ * 
+ * WINDOWED operation
+ * ==================
+ *
+ *    wdt_setup_reboot(MIN_TIME, MAX_TIME);
+ *    wdt_start();
+ *
+ *    // MIN_TIME < VALID_TIME < MAX_TIME
+ *    while (1) {
+ *        xtimer_usleep(VALID_TIME);
```

Comment:
```
 *        xtimer_usleep(MIN_TIME);
```

Reply:
this is still wrong

Reply:
Why? The window is `[MIN_TIME, MAX_TIME]`. If `kick()` before `MIN_TIME` it would be too early, but it works with whatever value in the window. Kind of like this image from stm32 datasheet. Is it different for samd21? Am I missing something obvious?

![image](https://user-images.githubusercontent.com/23060007/59385209-2942cb00-8d64-11e9-9337-af1da60f7cf0.png)

Reply:
Maybe the "drawing" is confusing?

Reply:
I agree that it should be `xtimer_usleep(MIN_TIME)` - but it's still `xtimer_usleep(VALID_TIME)` in the documentation.

Reply:
Ok, I see, the I think it comes to the drawing. I was using `VALID_TIME` as  `MIN_TIME < VALID_TIME < MAX_TIME`, but the drawing makes it look as if it is `0 < VALID_TIME < MIN_TIME `. Maybe something like this would be more clear?

```
 *
 *
 * 0(ms)                 MIN_TIME(ms)                MAX_TIME(ms)
 *  |-------------------------|-----------WINDOW----------|
 *
 * 0(ms)                        VALID_TIME(ms)
 *  |---------------------------------|
 *                                    ^
 *                                    |
 *                               wdt_kick()
```

Reply:
But `VALID_TIME` should not start before `MIN_TIME` is over.

I think the confusion stems from me thinking of `VALID_TIME` as a period that starts at `MIN_TIME` and ends at `MAX_TIME`. It's not a single point in time but a range.

Reply:
Ok How about if we use your drawing and have: `xtimer_usleep(MIN_TIME+VALID_TIME)` ?

Reply:
I'm fine with that. But having `VALID_TIME` there to me looks like it's a constant just like `MIN_TIME`, when it's not. It's `xtimer_usleep(MIN_TIME + x)`, where `x < MAX_TIME - MIN_TIME`.

Reply:
Ok I think that clearer, thanks for the feedback :)

Reply:
@benpicco I pushed according to your suggestions, does it look good now?

Reply:
Just add another line of comment so we have `x` 'defined' in both cases and we're good. :)